app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala

```diff
@@ -84,24 +95,43 @@ class RegisterController @Inject()(val messagesApi: MessagesApi,
         Logger.error(s"Could not get authorisation code: $e. Error description was" +
           s"${error_description.getOrElse("-")}, error code was ${error_code.getOrElse("-")}")
         // TODO: do something better
-        Action { InternalServerError(s"Could not get authorisation code: $e") }
+        Action {
+          InternalServerError(s"Could not get authorisation code: $e")
+        }
 
       case _ ⇒
         // we should never reach here - we shouldn't have a successful code and an error at the same time
         Logger.error("Inconsistent result found when attempting to retrieve an authorisation code")
-        Action { InternalServerError("") }
-
+        Action {
+          InternalServerError("")
+        }
     }
 
   def getCreateAccountHelpToSavePage: Action[AnyContent] = authorisedForHtsWithConfidence {
     implicit request ⇒
       Future.successful(Ok(views.html.register.create_account_help_to_save()))
   }
 
+  def validateOutGoingUserInfo(userInfo: Option[NSIUserInfo]): Either[String, Option[NSIUserInfo]] = {
+    val conf = app.configuration
+    userInfo match {
+      case None => Right(None)
```

Comment:
It turns out it does have to be a right. There is some other code that catches this situation and deals with it correctly. If I return a left here that code is never executed. Right(None) is the appropriate return.